tests/integration/test_charm.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test slurmrestd charm against other SLURM charms in the latest/edge channel."""
+
+import asyncio
+import logging
+import pathlib
+import pytest
+import tenacity
+
+from helpers import (
+    get_slurmctld_res,
+    get_slurmd_res,
+)
+from pytest_operator.plugin import OpsTest
+from typing import Any, Coroutine
+
+logger = logging.getLogger(__name__)
+
+SERIES = ["focal"]
+SLURMCTLD = "slurmctld"
+SLURMD = "slurmd"
+SLURMDBD = "slurmdbd"
+SLURMRESTD = "slurmrestd"
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+@pytest.mark.parametrize("series", SERIES)
+async def test_build_and_deploy(
+    ops_test: OpsTest, slurmrestd_charm: Coroutine[Any, Any, pathlib.Path], series: str
+    ) -> None:
```

Comment:
Formatting - you should not have an indent here.

```suggestion
) -> None:
```